tests/build/build.go

```diff
@@ -116,32 +121,23 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build-ser
 				Expect(f.SandboxController.DeleteUserSignup(f.UserName)).To(BeTrue())
 			}
 
-			switch gitProvider {
-			case "github":
-				// Delete new branches created by PaC and a testing branch used as a component's base branch
-				err = f.AsKubeAdmin.CommonController.Github.DeleteRef(helloWorldComponentGitSourceRepoName, pacBranchName)
-				if err != nil {
-					Expect(err.Error()).To(ContainSubstring("Reference does not exist"))
-				}
-				err = f.AsKubeAdmin.CommonController.Github.DeleteRef(helloWorldComponentGitSourceRepoName, componentBaseBranchName)
-				if err != nil {
-					Expect(err.Error()).To(ContainSubstring("Reference does not exist"))
-				}
-				err = f.AsKubeAdmin.CommonController.Github.DeleteRef(helloWorldComponentGitSourceRepoName, constants.PaCPullRequestBranchPrefix+customBranchComponentName)
-				if err != nil {
-					Expect(err.Error()).To(ContainSubstring("Reference does not exist"))
-				}
-				// Delete created webhook from GitHub
-				Expect(build.CleanupWebhooks(f, helloWorldComponentGitSourceRepoName)).ShouldNot(HaveOccurred())
-			case "gitlab":
+			err = gitClient.DeleteBranch(helloWorldRepository, pacBranchName)
+			if err != nil {
+				Expect(err.Error()).To(Or(ContainSubstring("Reference does not exist"), ContainSubstring("404")))
+			}
+			err = gitClient.DeleteBranch(helloWorldRepository, componentBaseBranchName)
+			if err != nil {
+				Expect(err.Error()).To(Or(ContainSubstring("Reference does not exist"), ContainSubstring("404")))
+			}
+
+			if gitProvider == "gitlab" {
```

Comment:
Better to avoid such things if you already use interface

Reply:
Please see https://github.com/konflux-ci/e2e-tests/pull/1421#discussion_r1795075570